rpcs3/Emu/Cell/Modules/sceNp2.cpp

```diff
@@ -613,25 +613,34 @@ error_code sceNpMatching2SignalingGetConnectionInfo(
 
 	switch (code)
 	{
-		case 1:
+		case SCE_NP_SIGNALING_CONN_INFO_RTT:
 		{
 			connInfo->rtt = 20000; // HACK
 			break;
 		}
-		case 2:
+	    case SCE_NP_SIGNALING_CONN_INFO_BANDWIDTH:
 		{
 			connInfo->bandwidth = 10'000'000; // 10 MBPS HACK
 			break;
 		}
-		case 5:
+	    case SCE_NP_SIGNALING_CONN_INFO_PEER_ADDRESS:
 		{
 			auto& sigh = g_fxo->get<named_thread<signaling_handler>>();
 			const auto si = sigh.get_sig2_infos(roomId, memberId);
 			connInfo->address.port = std::bit_cast<u16, be_t<u16>>(si.port);
 			connInfo->address.addr.np_s_addr = si.addr;
 			break;
 		}
-		case 6:
+	    case SCE_NP_SIGNALING_CONN_INFO_MAPPED_ADDRESS:
+		{
+			// TODO: wrong?
```

Comment:
It is indeed wrong but I didn't plan for it when designing the signaling handler sadly.
Mapped address is your own IP as seen from that peer.
The only real solution would be to change signaling_packet to include IP & Port it's sending the packet to(as the actual IP&port you receive it to are probably targeted at local IP behind router).